api/connector/simpleconnector.go

```diff
@@ -51,17 +51,26 @@ func NewSimple(opts SimpleConfig, dialOptions ...api.DialOption) (*SimpleConnect
 		Addrs:    opts.ControllerAddresses,
 		CACert:   opts.CACert,
 		ModelTag: names.NewModelTag(opts.ModelUUID),
+	}
+
+	isClientCredentialsProvided := opts.ClientID != "" && opts.ClientSecret != ""
```

Comment:
should we rather reverse the logic and assign the Tag field only when opts.Username is not empty? the terraform provider enforces that either username/password or clientID/clientSecret are defined (but not both)..

Reply:
Done. We're now checking `opts.Username` and decide on what to do.